tests/templates/test_embeddings/test_displacement_emb.py

```diff
@@ -0,0 +1,271 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Unit tests for the DisplacementEmbedding template.HE
+"""
+import pytest
+import numpy as np
+from pennylane import numpy as pnp
+import pennylane as qml
+
+
+class TestDecomposition:
+    """Tests that the template defines the correct decomposition."""
+
+    @pytest.mark.parametrize("features", [[1, 2, 3], [-1, 1, -1]])
+    def test_expansion(self, features):
+        """Checks the queue for the default settings."""
+
+        op = qml.templates.DisplacementEmbedding(features=features, wires=range(3))
+        tape = op.expand()
+
+        assert len(tape.operations) == len(features)
+        for idx, gate in enumerate(tape.operations):
+            assert gate.name == "Displacement"
+            assert gate.parameters[0] == features[idx]
+
+    def test_state_execution_amplitude(self):
+        """Checks the state using the amplitude execution method."""
+
+        features = np.array([0.1, 1.2])
+        n_wires = 2
+        dev = qml.device("default.gaussian", wires=n_wires)
+
+        @qml.qnode(dev)
+        def circuit(x=None):
+            qml.templates.DisplacementEmbedding(
+                features=x, wires=range(n_wires), method="amplitude", c=1.0
+            )
+            return [
+                qml.expval(qml.NumberOperator(wires=0)),
+                qml.expval(qml.NumberOperator(wires=1)),
+            ]
+
+        assert np.allclose(circuit(x=features), [0.01, 1.44], atol=0.001)
```

Comment:
We need a better test case here - it is dodgy to check a precomputed outcome. I left a todo.